src/containers/treePage/index.tsx

```diff
@@ -180,6 +181,19 @@ const TreePage: React.FC<TreeProps> = ({
       });
   };
 
+  const onClickForceUnadopt = () => {
+    protectedApiClient
+      .forceUnadoptSite(id)
+      .then(() => {
+        message.success('Force unadopted site!');
+        dispatch(getSiteData(id));
+        dispatch(getAdoptedSites());
+      })
+      .catch((err) => {
+        message.error(`Failed to force unadopt site: ${err.response.data}`);
```

Comment:
💭 Would it be worth our time to do a little refactoring at some point to move some of these functions into TreeInfo or even further down? It seems like a lot of them just rely on the dispatch function from `useDispatch` and the site id from lines 125/126 - maybe we could utilize a React context here to reduce how much prop drilling we do.

Reply:
I think the idea behind this is that `treePage` is supposed to be a "smart" component, while the components it renders are "dumb" components (more info [here](https://medium.com/@thejasonfile/dumb-components-and-smart-components-e7b33a698d43) and [here](https://javascript.plainenglish.io/react-all-about-components-35650a02ff50) but it's essentially that the page-level component has all of the logic so that all of the components within the page just take in everything and all they have to do is render). This does clutter the "smart" component quite a bit and I can't say I'm fully onboard with this concept but I think at this point, we can just leave it because we do this on p much every page and it'll probably take too long. Great catch tho!